allennlp/data/data_loaders/multi_process_data_loader.py

```diff
@@ -84,13 +84,20 @@ class MultiProcessDataLoader(DataLoader):
         the `reader` needs to implement
         [`manual_multi_process_sharding`](/api/data/dataset_readers/dataset_reader/#datasetreader).
 
+        !!! Warning
+            Multi-processing code in Python is complicated! We highly recommend you read the short
+            [Best practices](#multiprocessdataloader.best_practices) and
+            [Common issues](#multiprocessdataloader.common_issues) sections below before using this option.
+
     max_instances_in_memory: `int`, optional (default = `None`)
         If not specified, all instances will be read and cached in memory for the duration
         of the data loader's life. This is generally ideal when your data can fit in memory
         during training. However, when your datasets are too big, using this option
         will turn on lazy loading, where only `max_instances_in_memory` instances are processed
         at a time.
 
+        When using this option you should generally set it to a multiple of `batch_size`.
```

Comment:
Does that really matter? Isn't it just the size of the `pool` during shuffling?

Reply:
It matters when you're using a `BatchSampler`. Otherwise you're right, it's just the size of the pool used in `shuffle_iterable`. I should clarify.

Reply:
So it matters for how well shuffled your data is. If you set this to 1x batch size, you get no shuffle at all.